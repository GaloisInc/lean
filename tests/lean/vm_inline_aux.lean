namespace ex
set_option trace.compiler.optimize_bytecode true

@[inline] def {u} cond {a : Type u} : bool → a → a → a
| tt x y := x
| ff x y := y

-- cond should be inlined here
def foo (x : bool) :=
100 + cond x (10*10) (20*20)
end ex
