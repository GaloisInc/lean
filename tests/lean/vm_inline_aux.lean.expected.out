[compiler.optimize_bytecode]  ex.cond._main 4
0: move 2
1: cases2 4
2: move 0
3: ret
4: move 1
5: ret
[compiler.optimize_bytecode]  ex.cond 1
0: scnstr #0
1: closure ex.cond._main 1
2: ret
[compiler.optimize_bytecode]  ex.foo 1
0: move 0
1: cases2 6
2: scnstr #20
3: scnstr #20
4: cfun nat.mul
5: goto 9
6: scnstr #10
7: scnstr #10
8: cfun nat.mul
9: scnstr #100
10: cfun nat.add
11: ret
